mavros/6bb2672.bug

```diff
@@ -45,3 +45,8 @@ fix:
   languages:
     - C++
   time: 2017-03-30 (00:00)
+time-machine:
+  ros_distro: kinetic
+  ros_pkgs:
+    - mavros
+  datetime: 2017-03-30
```

Comment:
Is this precise enough?

`rosdistro` receives a lot of commits on a single day.

If you use the [get_issue_creation_date.py](https://github.com/rosin-project/rosinstall_generator_time_machine/blob/245fcd6525e8dc302273885041ac9921b1352474/get_issue_creation_date.py) script, it should give you a full timestamp which would be more precise.

Reply:
Doesn't `get_issue_creation_date.py` assume that there's an issue for the bug? A number of the mavros bugs don't have an issue.

Reply:
Also, I'm having a few issues providing more specific ISO 8601 timestamps. When I pass `2015-12-04T02:20:03+03`, I get the following error:
```
executing command: rosinstall_generator_tm.sh 2015-12-03 23:20:03 c6791f0 indigo mavros /home/chris/bugs/robust/mavros/c6791f0/deps.rosinstall
Already on 'master'
Your branch is up-to-date with 'origin/master'.
Going back to '2015-12-03 23:20:03'
Provided date '2015-12-03 23:20:03' is not a valid date ..
````

Reply:
> 
> 
> Doesn't `get_issue_creation_date.py` assume that there's an issue for the bug? A number of the mavros bugs don't have an issue.

No, it actually works for every resource on github that has a "created_at" property, which includes PRs as well. If it doesn't work, please report that.

Reply:
> 
> 
> Also, I'm having a few issues providing more specific ISO 8601 timestamps. When I pass `2015-12-04T02:20:03+03`, I get the following error:
> 
> ```
> executing command: rosinstall_generator_tm.sh 2015-12-03 23:20:03 c6791f0 indigo mavros /home/chris/bugs/robust/mavros/c6791f0/deps.rosinstall
> Already on 'master'
> Your branch is up-to-date with 'origin/master'.
> Going back to '2015-12-03 23:20:03'
> Provided date '2015-12-03 23:20:03' is not a valid date ..
> ```

I'm a bit confused by the command line you show: there appears to be a space between the date and the time, so it effectively becomes two args. That won't work.

It also doesn't include timezone info, perhaps that is what trips up `date`

Reply:
Yeah, that is strange. The time machine appears to be transforming the timestamp to UTC. I'll manually convert the stamps to UTC and see what happens.

Reply:
But what's up with the whitespace in the command line?

Reply:
Ah, my bad. I see the problem. The datetime is loaded as a `datetime` object in Python, which is pretty printed when it's added to the command string. I forgot that YAML would parse the datetime. I'll add a step to format the datetime correctly.